libraries/classes/Html/Generator.php

```diff
@@ -340,9 +351,7 @@ public static function getFunctionsForField(array $field, $insert_mode, array $f
         $functions = $dbi->types->getAllFunctions();
         foreach ($functions as $function) {
             $retval .= '<option';
-            if (isset($foreignData['foreign_link']) && $foreignData['foreign_link'] !== false
-                && $default_function === $function
-            ) {
+            if ($function === $default_function && ! isset($foreignData['foreign_field'])) {
```

Comment:
```suggestion
            if ($function === $default_function && ! isset($foreignData['foreign_field']) && $insert_mode) {
```
For me this would be the right operation but it would not match "Functions selected by default when inserting/changing row" from the documentation